conda_build/utils.py

```diff
@@ -718,7 +721,128 @@ def __exit__(self, et, ev, tb):
         # implicit return of None => don't swallow exceptions
 
 
+def get_installed_packages(path):
+    '''
+    Scan all json files in 'path' and return a dictionary with their contents.
+    Files are assumed to be in 'index.json' format.
+    '''
+    installed = dict()
+    for filename in glob(os.path.join(path, 'conda-meta', '*.json')):
+        with open(filename) as file:
+            data = json.load(file)
+            installed[data['name']] = data
+    return installed
+
+
+def _convert_lists_to_sets(_dict):
+    for k, v in _dict.items():
+        if hasattr(v, 'keys'):
+            _dict[k] = HashableDict(_convert_lists_to_sets(v))
+        elif hasattr(v, '__iter__'):
+            _dict[k] = sorted(list(set(v)))
+    return _dict
+
+
+class HashableDict(dict):
+    """use hashable frozen dictionaries for resources and resource types so that they can be in sets
+    """
+    def __init__(self, *args, **kwargs):
+        super(HashableDict, self).__init__(*args, **kwargs)
+        self = _convert_lists_to_sets(self)
+
+    def __hash__(self):
+        return hash(json.dumps(self, sort_keys=True))
```

Comment:
> It is enabled with the -R flag. The change was added in for 2.7.3rc1.
https://hg.python.org/cpython/raw-file/v2.7.3/Misc/NEWS

Didn't know setting `PYTHONHASHSEED` would enable this on Python 2. Good to know. Nice little check for moving to Python 2/3 or Python 3 compatible code. Thanks.